feat: list of artists

KindImagination opened the pull request:
closes #322 

How to check:
1) Open the artists page
 2) You can see the artists 
 3) You can filter by city and language.

KindImagination committed:
feat: enhance artist profile display and implement pagination for artist list

- Updated ArtistCard component to include dynamic avatar URLs and display styles with a count for additional styles.
- Modified ArtistsView component to fetch artist data from the API with pagination support, including loading and error states.
- Added a new useArtists query in trpc for fetching artist profiles.
- Updated profile schema to accommodate structured styles data.
- Implemented server-side pagination in the profiles router for artist queries.

KindImagination committed:
feat: enhance artist language and location handling in components

- Updated ArtistCard and ArtistsView components to utilize computed properties for extracting languages from both languages array and locales object.
- Improved location display logic to include city name where applicable.
- Modified profile schema to include city information and ensure locales are properly defined.
- Adjusted filtering logic in ArtistsView to account for languages from both sources.

KindImagination committed:
fix: update artist link format in ArtistCard component

razbakov commented on components/ArtistsView.vue:
```diff
@@ -1,9 +1,76 @@
+    result.value = response
+    // If we have more items than our limit, assume there are more to load
+    hasMore.value = response.length >= limit.value
+    console.log('Artists data:', result.value)
```
console.log shouldn't be in production

KindImagination replied:
removed

razbakov commented on components/ArtistsView.vue:
```diff
@@ -1,9 +1,76 @@
+    hasMore.value = response.length >= limit.value
+    console.log('Artists data:', result.value)
+  } catch (e) {
+    console.error('API error:', e)
```
not like that, use toast

KindImagination replied:
changed to with toast

razbakov commented on components/ArtistsView.vue:
```diff
@@ -1,9 +1,76 @@
+})
 
-const artists = mockArtists
+// Access artists data safely
```
remove obvious comments

razbakov commented on components/ArtistsView.vue:
```diff
@@ -82,10 +149,10 @@ const specialtiesOptions = computed(() => {
       .filter(
         (artist) =>
-          artist.roles.includes('dj') || artist.roles.includes('videographer')
+          artist.roles?.includes('dj') || artist.roles?.includes('videographer')
```
filters should be done in backend, not in frontend

KindImagination committed:
Remove unnecessary console logs

KindImagination committed:
feat: add error handling for artist data fetching in ArtistsView

- Implemented toast notifications for error handling when fetching artists and loading more results.
- Enhanced user feedback by displaying error messages in case of API failures.

KindImagination committed:
bugfix

KindImagination committed:
refactor: clean up comments in ArtistCard and ArtistsView

KindImagination committed:
feat: enhance artist filtering and loading logic in ArtistsView

- Implemented dynamic query parameters for fetching artists based on selected filters (role, location, language, and search query).
- Added loading states for locations and languages, improving user experience during data fetching.
- Introduced debounce functionality for the search input to optimize performance.
- Updated the UI to reflect loading states and disabled filters where applicable.
- Refactored the filtering logic to ensure accurate results based on user selections.

razbakov commented on server/trpc/routers/profiles.ts:
```diff
@@ -268,4 +268,151 @@ export const profilesRouter = router({
+          role: z.string().optional(),
+          location: z.string().optional(),
+          language: z.string().optional(),
+          search: z.string().optional(),
```
this is usually called `q` or `query`

razbakov commented on server/trpc/routers/profiles.ts:
```diff
@@ -268,4 +268,151 @@ export const profilesRouter = router({
+      }
+
+      if (input?.language && input.language !== 'all') {
+        // Use JsonPath to check if the language key exists in the locales object
```
please remove the comments

KindImagination committed:
refactor: clean up comments and improve profile update schema in profiles router

- Removed outdated comments and unnecessary code snippets for clarity.
- Streamlined the profile update schema by adjusting optional fields and ensuring consistency.
- Enhanced readability of the filtering logic in the profiles router.

KindImagination committed:
refactor: clean up comments and improve readability in ArtistCard and ArtistsView

- Removed outdated comments and unnecessary code snippets for clarity.
- Enhanced the structure of computed properties for better readability.
- Streamlined the logic for fetching artists and handling filters in ArtistsView.

KindImagination committed:
refactor: rename search parameter to query in profiles router

KindImagination committed:
bugfix
